Spanner/src/Transaction.php

```diff
@@ -615,6 +648,19 @@ public function commit(array $options = [])
             throw new \BadMethodCallException('The transaction cannot be committed because it is not active');
         }
 
+        if (is_array($this->transactionId) and isset($this->transactionId['begin'])) {
+            unset($this->options['begin']);
+            $res = $this->operation->transaction($this->session, $this->options+[
+                'fromTransaction' => true
+            ]);
+            if (is_array($res) and isset($res['id'])) {
+                $this->transactionId = $res['id'];
+            } else {
+                // singleUse transaction
+                $this->transactionId = null;
+            }
```

Comment:
we seem to be alternating between `array`, `string`, and `null` types for `$transactionId`. This is not good practice.

Reply:
Thank you for the suggestion, we can create a new attribute then